src/logger.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+const KafkaProducer = require('no-kafka');
+const configFunctions = require('../src/config');
+const config = configFunctions.getConfig();
+
+let producer;
+const initProducer = (errorCallBack = console.error) => {
+  try {
+    producer = new KafkaProducer.Producer({
+      connectionString: config.connectionString,
+      ssl: {
+        cert: config.sslCert,
+        key: config.sslKey,
+      },
+    });
+    producer.init();
+  } catch (err) {
+    errorCallBack(`Failed to initialized producer error: ${err}`);
+  }
+};
+
+const logFunc = {
+  info: console.log,
+  warn: console.warn,
+  error: console.error,
+  trace: console.log,
+  debug: console.log,
+};
+
+const writeLocalLog = (logMessage) => {
+  logFunc[logMessage.message.key](logMessage.message.value);
+};
+
+const logger = {
+  error: (value) => writeLog(value, 'error'),
+  warn: (value) => writeLog(value, 'warn'),
+  info: (value) => writeLog(value, 'info'),
+  debug: (value) => writeLog(value, 'debug'),
+  silly: (value) => writeLog(value, 'trace'),
+};
+
+const writeLog = (value, key = 'info', topic = 'refocus-whitelist',
+                  callback = console.warn) => {
+  const messageValue = {
+    sendTimeStamp: new Date(),
+    value,
+  };
+  const logMessage = {
+    topic,
+    partition: 0,
+    message: {
+      key,
+      value: JSON.stringify(messageValue),
+    },
+  };
+  if (configFunctions.kafkaLogging) {
+    producer.send(logMessage).catch(err => {
+      callback('Sending the log message to Kafka cluster failed, ' +
+      `writing locally, error: ${err}`);
+      writeLocalLog(logMessage);
```

Comment:
Is it necessary to log the full message on failure? I would think logging the error would be enough, and It seems wrong to log it locally when local logging is disabled.

Reply:
But what if the send is failed (since it is a network call?) I thought we wouldn't wanna lose the logMessage even if local logging is off.

Reply:
Most of the time we're more concerned with overall stats than any individual log line, and in the hopefully rare case that a log line doesn't go through, logging it locally won't help with that anyway because we will be aggregating from the kafka side.
@iamigo thoughts?

Reply:
Talked with Ian, what you said definitely makes sense. He suggested in the rare case it fails, we can just retry it? I guess it does make sense to not log if localLogging is set to off explicitly. Will make the change.